### PR TITLE
ref: Remove old TODOs

### PR DESCRIPTION
**Pull request by jan-auer**
Removes some old TODOs. Most notably, this removes the `_relay_processed` marker from event payloads.

### CODE REVIEW

**Comment by jan-auer** [comment]
@untitaker can we remove this aswell?

https://github.com/getsentry/relay/blob/a1b2f4c5879c25b29c8fe5c49da8589b087748cd/relay-general/src/protocol/span.rs#L37-L40

**Review by untitaker — APPROVED** [review]

File: relay-general/src/pii/compiledconfig.rs [hunk 1]
```diff
@@ -38,6 +39,7 @@ fn get_rule(config: &PiiConfig, id: &str) -> Option<RuleRef> {
     }
 }
 
+#[allow(clippy::mutable_key_type)]
```
Comment (jan-auer) [inline relay-general/src/pii/compiledconfig.rs:42]: This seems to be a false-positive, I'm assuming this is due to `Regexes` stored inside `RuleRef`. See https://rust-lang.github.io/rust-clippy/master/index.html#mutable_key_type

**Comment by untitaker** [comment]
Let's do that not just yet. I thought about this recently but then we started another schema discussion like two days ago

**Review by untitaker — APPROVED** [review]

**Review by untitaker — APPROVED** [review]

**Comment by untitaker** [comment]
@jan-auer your consolidation broke our Python 2 build, as black does not install under Python 2.

**Pull request merged**